Server/Cordova/serverConfig.txt
<!DOCTYPE HTML>
<html>
<head>
	<title>OVERSIGHT</title>
	<link rel="shortcut icon" href="imgs/Logo3.png">

	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
	<!--Let browser know website is optimized for mobile-->
	<meta name="viewport" content="width=device-width, initial-scale=1.0"/>

	<!--Import Google Icon Font-->
	<link href="css/material+icons.css" rel="stylesheet">

	<!--Import materialize.css-->
	<link type="text/css" rel="stylesheet" href="css/materialize.css"  media="screen,projection"/>
	<style>
		@viewport {
		  width: device-width ;
		  zoom: 1.0 ;
		}
		@-ms-viewport {
		  width: device-width ;
		}
		*{
			margin:0;
			font-family: verdana,arial;
			font-smooth: always;
			font-smoothing: antialiased;
			font-size:16px;
			user-select:none;
			color:black;
		}
		body{
			width: 100%;
			height: 100%;
			background: rgba(138,43,226,1) radial-gradient(transparent,rgba(75,0,130,1));
			background: rgba(138,43,226,1) linear-gradient(rgba(75,0,130,1),rgba(138,43,226,1));
			background-attachment: fixed;
			background-position: center;
			background-size: 100% 100%;
			color: #fff;
			align-items: center;
			animation: fadein .7s;
		}

		@keyframes fadein {
				from { opacity: 0; }
				to   { opacity: 1; }
		}
		.appLogo{
			margin:0 auto;
			height:100px;
			width:100px;
			background:url("imgs/Logo_100px.png") no-repeat;
			background-size:cover;
			font-size: 3.2vh;
			user-select:none;
			position:relative;
		}
		.appName{
			font-size: 2rem;
			color:white;
			font-weight:bold;
			margin: 0 auto;
			width: fit-content;
			user-select:none;
			position:relative;
		}
		.submitbtn{
			width:100%;
			font-weight:bold;
			font-size:20px;
			margin:10px 0;
			box-sizing:border-box;
		}
		.spacing1{
			display:inline-block;
			width:100%;
			padding: 10px;
			margin: 0.1rem 0 1.5rem 0;
			position:relative;
		}
		.modalContents{
			width: 100%;
			height: 100%;
			display:block;
			user-select: none
		}

		input.border_bottom_green.valid:not([type]),
		input.border_bottom_green.valid:not([type]):focus,
		input[type=text].border_bottom_green.valid:not(.browser-default),
		input[type=text].border_bottom_green.valid:not(.browser-default):focus,
		input[type=password].border_bottom_green.valid:not(.browser-default),
		input[type=password].border_bottom_green.valid:not(.browser-default):focus,
		input[type=email].border_bottom_green.valid:not(.browser-default),
		input[type=email].border_bottom_green.valid:not(.browser-default):focus,
		input[type=url].border_bottom_green.valid:not(.browser-default),
		input[type=url].border_bottom_green.valid:not(.browser-default):focus,
		input[type=time].border_bottom_green.valid:not(.browser-default),
		input[type=time].border_bottom_green.valid:not(.browser-default):focus,
		input[type=date].border_bottom_green.valid:not(.browser-default),
		input[type=date].border_bottom_green.valid:not(.browser-default):focus,
		input[type=datetime].border_bottom_green.valid:not(.browser-default),
		input[type=datetime].border_bottom_green.valid:not(.browser-default):focus,
		input[type=datetime-local].border_bottom_green.valid:not(.browser-default),
		input[type=datetime-local].border_bottom_green.valid:not(.browser-default):focus,
		input[type=tel].border_bottom_green.valid:not(.browser-default),
		input[type=tel].border_bottom_green.valid:not(.browser-default):focus,
		input[type=number].border_bottom_green.valid:not(.browser-default),
		input[type=number].border_bottom_green.valid:not(.browser-default):focus,
		input[type=search].border_bottom_green.valid:not(.browser-default),
		input[type=search].border_bottom_green.valid:not(.browser-default):focus,
		textarea.materialize-textarea.border_bottom_green.valid,
		textarea.materialize-textarea.border_bottom_green.valid:focus,
		.select-wrapper.border_bottom_green.valid > input.border_bottom_green.select-dropdown{
		  border-bottom: 1px solid #4CAF50;
		  -webkit-box-shadow: 0 1px 0 0 #4CAF50;
				  box-shadow: 0 1px 0 0 #4CAF50;
		}

		.text_indent{
			text-indent:2rem
		}
		input.border_bottom_darkgray:not([type]):focus:not([readonly]),
		input[type=text].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=password].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=email].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=url].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=time].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=date].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=datetime].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=datetime-local].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=tel].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=number].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		input[type=search].border_bottom_darkgray:not(.browser-default):focus:not([readonly]),
		textarea.border_bottom_darkgray.materialize-textarea:focus:not([readonly]) {
		  border-bottom: 1px solid #616161;
		  -webkit-box-shadow: 0 1px 0 0 #757575;
				  box-shadow: 0 1px 0 0 #757575;
		}
		html, body, .my-wrapper {
		  height: 100%;
		}
		.border_right{
			border-right:solid 1px white;
			height:95vh;
		}
		.modal-limits{
			overflow-x:hidden;
		}

		#toast-container {
		  top: auto !important;
		  right: auto !important;
		  bottom: 50%;
		}
	</style>
	<!--JavaScript at end of body for optimized loading-->

	<script type="text/javascript" src="js/jquery-3.2.1.min.js"></script>
  <script type="text/javascript" src="js/materialize.js"></script>
	<script type="text/javascript" src="js/socket.io.js"></script>

	<!--cordova-->

	<script>
		function mySubmit(e) { 
		  e.preventDefault(); 
		  try {
		   
		  } catch (er) {
		  
		  }
		  return false;
		}
	
		$(document).ready(function(){
			// the "href" attribute of .modal-trigger must specify the modal ID that   wants to be triggered

			function modalClosed(){
				$('input').each(function(){
				  $(this).val("");
				  $(this).removeClass('invalid');
				  $(this).removeClass('valid');
				});
				
				$('label').each(function(){
					$(this).removeClass('active');
				})
			}
			
			function testHost(devHost,type){
				$.ajax({
					type: "POST",
					url: "http://" + devHost + "/check.php",
					data: "none",
					crossDomain: true,
					contentType: "application/x-www-form-urlencoded; charset=utf-8",
					success: function(data) {
						if(type=="set"){
							window.localStorage.setItem('I-O-T',deviceHost);
						}
						window.location.href = "index.html";
					},
					error: function(XMLHttpRequest, textStatus, errorThrown) {
						SendToast("Connection Failed! Host Unreachable!");
					}
				});
			
			}

			$('.modal').modal({onCloseEnd: function() { modalClosed(); }});
			
			$('#saveConfig').click(function(){
				if($('#serverAddress').val().trim()!=""){
					deviceHost = $('#serverAddress').val().trim();
					testHost(deviceHost,"set");
				}
			});
			
			$('#backbutton').click(function(){
				var ip = window.localStorage.getItem('I-O-T');
				if(ip){
					testHost(ip,"cancel");
				}else{
					$('#serverAddress').addClass('invalid');
					SendToast("You have no server address. It is required to proceed");
				}
			});
			
			function SendToast(message){
				$('.toast').hide();

				var toastHTML = "<span style='color: white; word-break: keep-all;  width: 70%; font-size: 1em;'>" +message+ "</span><button style='color: grey; width: 30%;' class='btn-flat toast-action'>Close</button>";
				M.toast({
					html: toastHTML
				});

				$('.toast-action').click(function() {
					M.Toast.dismissAll();
				});
			}
			$('input').on('input',function(){
				$(this).removeClass('invalid');
				$(this).removeClass('valid');
			});
		});
	</script>
</head>
<body>
	<div class="my-wrapper valign-wrapper">
		<div class="container">
			<div class="row">
				<div class="col l8 s12 hide-on-med-and-down border_right">
					<h5 class="white-text">Company Bio</h5>
					<p class="grey-text text-lighten-4">We are a team of college students working on this project like it's our full time job. Any amount would help support and continue development on this project and is greatly appreciated.</p>
				</div>

				<div class="col l4 s12 show-on-medium-and-down">
					<div class="container">
						<div class="row">
								<div class="appLogo"></div>
								<p class="appName ">OVERSIGHT</p>
							<form class="col s12" method="POST" action="index.html" id='serverForm' onsubmit="return mySubmit(event)">
								<div class="input-field col s12">
									<i class="material-icons prefix">insert_link</i>
									<input id="serverAddress" type="text" class="validate white-text" autocomplete="off" required/>
									<span class="helper-text" data-error="&#10006; This field is required!" data-success=""></span>
									<label for="serverAddress">SERVER ADDRESS</label>
								</div>
								<button id='saveConfig' class="submitbtn btn waves-effect waves-light green" type="submit" name="action">SAVE CONFIG</button>
								<a id="backbutton" class="submitbtn btn waves-effect waves-light  red" href="#Back">BACK</a>
							</form>

							<div id='sampleholder'></div>
						</div>
					</div>
				</div>
			</div>
		</div>
	</div>

	
</body>
</html>


Server/Cordova/js/settings.js
var settings = {
	socket: "",
	limit: "",
	deviceauthentication: "",
	price: 0.0,
	admin: ""
}
var admin_code = "";
var class_value = "";
var admin_password_class = "";
var inputprice = 0.0;
var inputpass = "";

function load_xml() {
	$.ajax({
		url: "http://" + deviceHost + "/xml/overpass.xml",
		dataType: "xml",
		crossDomain: true,
		contentType: "application/x-www-form-urlencoded; charset=utf-8",
		success: function (data) {
			xmlDoc = data;
			admin_code = xmlDoc.getElementsByTagName('admin')[0].firstChild.data;
			settings.admin = admin_code;
		}
	});
}

function initiate_settings() {
	load_xml();
	var label = $('#row_setting').attr('name');
	var logged_username = $('#username').val();
	settings.socket = $('#h_socket').val();
	settings.limit = $('#h_limit').val();
	settings.deviceauthentication = $('#h_authentication').val();
	settings.price = $('#h_price').val() * 1;
	$('#price_label').text(label);

	$('#s_socket').change(function () {
		settings.socket = (this.checked) ? 'true' : 'false';
		updateSettings();
	});

	$('#s_limit').change(function () {
		settings.limit = (this.checked) ? 'true' : 'false';
		updateSettings();
	});

	$('#s_authenticate').change(function () {
		$('#confirm_admin').attr('href','#');
		$('#confirm_admin').removeClass('triggerForPrice');
		var instance = M.Modal.getInstance($('#adminConfirmChangeRate'));
		instance.open();
	});
	
	$('#price-button-admin').focus(function(){
		$('#confirm_admin').attr('href','#ChangeMultiplyerRate');
		$('#confirm_admin').addClass('triggerForPrice');
	});

	$('#s_price').change(function () {
		inputprice = (this.value) * 1.0;
	});

	$('#btn_price').click(function () {
		if ($("#s_price").val().trim() != "") {
			settings.price = inputprice;
			$(this).addClass('modal-close');
			$("#s_price").val("");
			$("#price_label").removeClass('active');
			updateSettings();
		} else {
			$(this).removeClass('modal-close');
			inc_error("Price Value Required!");
		}
	});

	$('#new_password').keyup(function () {
		inputpass = $(this).val().trim();
	});

	$('#cancel').click(function () {
		admin_pass_reset();
	});

	$('.cancel-btns').click(function () {
		$('.p-fields').each(function () {
			$(this).val("");
		});

		$('.f-label').each(function () {
			$(this).removeClass("active");
		});

		settings.admin = admin_code;
		inputpass = "";
	});

	
//------------------------------------- PASSWORD -----------------------------------------------------------

	$('input').on('input',function(){
		$(this).removeClass('valid');
		$(this).removeClass('invalid');
	});
	
	$("#txtOldPassword").on('input',function(){
	if(this.value == admin_code){
	  $(this).addClass('valid');
	} else {
	  $(this).addClass('invalid');
	}
  });

  $("#txtNewPassword").on('input',function(){
	$("#txtConfirmPassword").removeClass('valid');
	if(this.value==$("#txtConfirmPassword").val()){
	  $("#txtConfirmPassword").addClass('valid');
	  inputpass = $(this).val();
	}
  });

  $("#txtConfirmPassword").on('input',function(){
	if(this.value==$("#txtNewPassword").val()){
	  $("#txtConfirmPassword").addClass('valid');
	  inputpass = $(this).val();
	} else {
	  $("#txtConfirmPassword").addClass('invalid');
	}
  });

  $(".save-password").focus(function(){
	if($("#txtOldPassword").hasClass("valid") &&  $("#txtNewPassword").hasClass("valid") && $("#txtConfirmPassword").hasClass("valid")){
	  $(this).addClass("modal-close");
	  settings.admin = inputpass;
	  updateSettings();
	} else {
	  inc_error("Invalid / Missing Field");
	  $(this).removeClass("modal-close");
	}
  });
  
	admin_password_class = $('#admin_password').attr('class');
	$('#validation_status').attr('data-error', "This Field is Required");
	$('#confirm_admin').focus(function () {
		if($(this).hasClass('triggerForPrice')){
			if ($('#admin_password').val().trim() == admin_code) {
				admin_pass_reset();
			} else {
				inc_error("Invalid Pass!");
			}
		} else {
			$('#confirm_admin').removeClass('modal-trigger');
			if ($('#admin_password').val().trim() == admin_code) {
				admin_pass_reset();
				if(settings.deviceauthentication.match(/false/i)){
					$('.myCheckbox').prop('checked', true);
					settings.deviceauthentication = 'true';
					updateSettings();
				} else if(settings.deviceauthentication.match(/true/i)){
					$('.myCheckbox').prop('checked', false);
					settings.deviceauthentication = 'false';
					updateSettings();
				}
			} else {
				inc_error("Invalid Pass!");
			}
		}
	});

	class_value = $('#confirm_admin').attr('class');
	$('#admin_password').keyup(function () {
		$('#validation_status').attr('data-error', "This Field is Required");
		$('#admin_password').removeClass("invalid");

		if ($('#admin_password').val() != "") {
			if ($('#admin_password').val() == admin_code) {
				$('#confirm_admin').addClass('modal-trigger modal-close');
			} else {
				$('#confirm_admin').removeClass('modal-trigger modal-close');
				$('#validation_status').attr('data-error', "Wrong Password");
			}
		}
	});
	
	function admin_pass_reset() {
		$('#passlabel').removeClass("active");
		$('#admin_password').val("");
		$('#admin_password').attr('class', admin_password_class);
		$('#admin_password').removeClass("invalid");
	}

	function checkSettings() {
		$.ajax({
			type: "POST",
			data: "getSettings=1",
			url: 'http://' + deviceHost + '/changeSettings.php',
			crossDomain: true,
			contentType: "application/x-www-form-urlencoded; charset=utf-8",
			success: function (data) {
				var db_settings = JSON.parse(data.trim());

				if (db_settings[0].socket == "true") {
					$('#s_socket').prop("checked", true);
				} else {
					$('#s_socket').prop("checked", false);
				}

				if (db_settings[0].limitation == "true") {
					$('#s_limit').prop("checked", true);
				} else {
					$('#s_limit').prop("checked", false);
				}

				if (db_settings[0].authentication == "true") {
					$('#s_authenticate').prop("checked", true);
				} else {
					$('#s_authenticate').prop("checked", false);
				}

				$('#price_label').text("Price (₱ " + db_settings[0].price + ")");

				settings.socket = db_settings[0].socket;
				settings.limit = db_settings[0].limitation;
				settings.deviceauthentication = db_settings[0].authentication;
				settings.price = db_settings[0].price;
				settings.admin = db_settings[0].admin;
				admin_code = db_settings[0].admin;

			},
			error: function (XMLHttpRequest, textStatus, errorThrown) {
				inc_error(errorThrown);
			}
		});
	}

	setInterval(checkSettings, 500);
}

function updateSettings() {
	var string = JSON.stringify(settings);
	var params = "settings=" + string;
	var xhr = new XMLHttpRequest();
	xhr.onreadystatechange = function () {
		if (xhr.readyState == 4 && xhr.status == 200) {
			console.log("Data Sent Successfully!");
			$('#confirm_admin').attr('class', class_value);
			$('#price_label').text("Price (₱ " + settings.price + ")");
			resetAllInputs();
			load_xml(); 
		}
	}
	xhr.open("POST", "http://" + deviceHost + "/changeSettings.php", true);
	xhr.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
	xhr.send(params);
	return false;
}

//-------------------------------------MINOR FUNCTIONS------------------------------------------------

	function inc_error(error) {
		$('.toast').hide();

		M.Toast.dismissAll();
		var toastHTML = "<span style='color: white; word-break: keep-all;  width: 70%; font-size:1em;'>"+error+"</span><button style='color: grey; width: 30%;' class='btn-flat toast-action'>Close</button>";
		M.toast({html: toastHTML});

		$('.toast-action').click(function(){
			 M.Toast.dismissAll();
		});
	}
	
	function resetAllInputs(){
		$('input').each(function(){
			$(this).val("");
			$(this).removeClass('valid');
			$(this).removeClass('invalid');
		});
		$('label').each(function(){
			$(this).removeClass('active');
		});
	}
//-------------------------------------------------------------------------------------------